src/sentry/rules/actions/base.py

```diff
@@ -83,56 +90,152 @@ def get_form_instance(self):
         return self.form_cls(self.data, integrations=self.get_integrations())
 
 
+def _linked_issues(event, integration):
+    return ExternalIssue.objects.filter(
+        id__in=GroupLink.objects.filter(
+            project_id=event.group.project.id,
+            group_id=event.group.id,
+            linked_type=GroupLink.LinkedType.issue,
+        ).values_list("linked_id", flat=True),
+        integration_id=integration.id,
+    )
+
+
+def get_linked_issue_ids(event, integration):
+    return _linked_issues(event, integration).values_list("key", flat=True)
+
+
+def has_linked_issue(event, integration):
+    return _linked_issues(event, integration).exists()
+
+
+def create_link(key, integration, installation, event):
+    external_issue = ExternalIssue.objects.create(
+        organization_id=event.group.project.organization.id,
```

Comment:
@ceorourke Nit: use `event.group.project.organization_id` here instead to avoid fetching the organization from the DB.